src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYearIndex = getIndex(fromFormat);
+  const newYearIndex = getIndex(toFormat);
+  const oldYearLength = fromFormat[oldYearIndex].length;
+  const newYearLength = toFormat[newYearIndex].length;
+  const oldJoin = fromFormat[3];
+  const newJoin = toFormat[3];
+  const newDate = date.split(oldJoin);
+  const reorderedDate = [];
+  const oldYear = newDate[oldYearIndex];
+
+  function getIndex(index) {
+    return index.indexOf('YY') !== -1
+      ? index.indexOf('YY')
+      : index.indexOf('YYYY');
+  }
+
+  function getYear(number, from, to) {
+    if (from > to) {
+      return number.slice(2);
+    }
+
+    if (from < to) {
+      return number < 30 ? '20' + number : '19' + number;
+    }
+
+    return number;
+  }
```

Comment:
In the getYear function, you are converting the number to a string in order to perform the slicing or concatenation operation. This can be avoided by using arithmetic operations which would be more efficient.